aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -421,7 +437,7 @@ Best practices are to:
 If a class seems to have too many injected dependencies, this is generally a sign that the class has too many responsibilities and is violating the [Single Responsibility Principle (SRP)](/dotnet/standard/modern-web-apps-azure-architecture/architectural-principles#single-responsibility). Attempt to refactor the class by moving some of its responsibilities into a new class. Keep in mind that Razor Pages page model classes and MVC controller classes should focus on UI concerns. Business rules and data access implementation details should be kept in classes appropriate to these [separate concerns](/dotnet/standard/modern-web-apps-azure-architecture/architectural-principles#separation-of-concerns).
 
 ### Disposal of services
-
+<!-- REVIEW: define user code.  -->
```

Comment:
I think this means something like this:

```csharp
var mySingleton = new MySingleton();
services.AddSingleton(mySingleton);
```
`mySingleton` isn't disposed by the container because _we_ provided the instance directly.

However, these two approaches result in automatic disposal:

```csharp
services.AddSingleton<MySingleton>();
services.AddSingleton(() => new MySingleton()); // Factory.
```

This one's more interesting:

```csharp
var mySingleton = new MySingleton();
services.AddSingleton(() => mySingleton);
```

This also causes automatic disposal, because the container assumes you created an instance on its behalf. I don't think you should ever really do this, though. It'd be dangerous if you tried to use `mySingleton` after the container was disposed.